feat: 탐험 중인 장소 조회 기능 구현

dain391 opened the pull request:
## 📌 개요
> 현재 탐험 중인 장소 조회 API를 구현

<img width="829" height="712" alt="image" src="https://github.com/user-attachments/assets/f30a9188-1797-4c63-b25d-aff781985f61" />

## ✅ 작업 사항
- [x] ExplorationRepository에 탐험 종료 시점 기준 탐험 조회 메서드 추가
- [x] ExplorationCurrentResponse DTO 정의
- [x] ExplorationService에 현재 탐험 조회 로직 추가
- [x] ExplorationController에 GET /api/exploration/current API 구현

## 🔍 리뷰 요청 포인트
- Exploration의 종료 조건 검증 로직
- 유효성 검증 누락된 부분 없는지 봐주세요.

## 💬 기타 사항
- 관련 이슈: #65

dain391 committed:
refactor: 탐험 패키지명 변경

- Exploration -> exploration

#65

dain391 committed:
feat: 현재 진행 중인 탐험 조회를 위한 Repository 메서드 추가

#65

dain391 committed:
feat: 현재 탐험 중인 장소 응답 DTO 생성

#65

dain391 committed:
feat: 현재 진행 중인 탐험 조회 서비스 로직 구현

- 현재 탐험 중인 탐험이 없는 경우의 에러 코드 추가(NO_ACTIVE_EXPLORATION)

#65

dain391 committed:
feat: 현재 탐험 중인 장소 조회 API 추가

#65

dain391 committed:
fix: Exploration 엔티티 필드명 오타 수정

- startedAt → startAt

#65

GyeongSe99 commented on src/main/java/com/explorer/gabom/domain/exploration/controller/ExplorationController.java:
```diff
@@ -33,4 +37,15 @@ public ResponseEntity<ApiResponse<ExplorationStartResponse>> startExploration(
 	}
+
+	// 탐험 중인 장소 조회
+	@PreAuthorize("isAuthenticated()")
```
요건 무슨 기능을 하나요?

gyogyo0212 replied:
오..

dain391 replied:
`@PreAuthorize("isAuthenticated()")`
로그인한 사용자만 접근할 수 있는 API입니다.
인증된 사용자만 해달 메서드를 실행하도록 제한합니다!

`@AuthenticationPrincipal`랑 같이 사용해서 로그인한 사용자 정보 가져올 때 쓰는 걸로 알고있어요...

GyeongSe99 replied:
현재 SecurityConfig에 이미 설정되어있는 기능이라서 빼주셔도 될 것 같습니다. 
auth외의 모든 페이지는 로그인 한 유저만 접근할 수 있도록 설정되어있어요

GyeongSe99 commented on src/main/java/com/explorer/gabom/domain/exploration/service/ExplorationService.java:
```diff
@@ -70,4 +72,15 @@ public ExplorationStartResponse startExploration(Long userId, Long placeId, Expl
+	// 탐험 중인 장소 조회
+	public ExplorationCurrentResponse getCurrentExploration(Long userId) {
+		Exploration exploration = explorationRepository
+			.findTopByUserIdAndEndAtAfterOrderByStartAtDesc(userId, LocalDateTime.now())
```
제가 이해한 바로는 정렬이 나중에 시작한 탐험이 제일 앞에 오도록 하는 정렬인 것 같은데 맞나요?
남은 시간을 본다고 생각한다면 남은 시간이 얼마 남지 않은 탐험이 제일 앞에오는게 맞는 것 같아서요

gyogyo0212 replied:
일리가 있는것같습니다..?

dain391 replied:
탐험 시작 시간(`startAt`)을 내림차순 정렬해서 가장 최근에 시작한 탐험을 조회하도록 만든거였습니다... 

근데 세경님 말이 맞는 듯!
`findTopByUserIdAndEndAtAfterOrderByEndAtAsc`로 수정할까요?

GyeongSe99 replied:
넵

dain391 replied:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

dain391 committed:
Merge branch 'develop' into feature/currentExploration

dain391 committed:
feat: @PreAuthorize("isAuthenticated()") 제거

#65

dain391 committed:
feat: 탐험 중인 장소 조회 시 남은 시간이 가장 적은 탐험 우선으로 정렬 기준 수정

- ExplorationRepository 메서드 정렬 기준을 startAt 기준 내림차순 → endAt 기준 오름차순으로 변경
- 남은 시간이 가장 적은 탐험이 먼저 조회되도록 개선

#65

dain391 committed:
import문 최적화

#65
